crates/ruff/src/rules/pycodestyle/rules/logical_lines/missing_whitespace_around_operator.rs

```diff
@@ -71,83 +71,77 @@ pub(crate) fn missing_whitespace_around_operator(
             _ => {}
         };
 
-        let needs_space = if kind == TokenKind::Equal && parens > 0 {
-            // Allow keyword args or defaults: foo(bar=None).
-            NeedsSpace::No
-        } else if kind == TokenKind::Slash {
-            // Tolerate the "/" operator in function definition
-            // For more info see PEP570
-
-            // `def f(a, /, b):` or `def f(a, b, /):` or `f = lambda a, /:`
-            //            ^                       ^                      ^
-            let slash_in_func = matches!(
-                tokens.peek().map(|t| t.kind()),
-                Some(TokenKind::Comma | TokenKind::Rpar | TokenKind::Colon)
-            );
-
-            NeedsSpace::from(!slash_in_func)
-        } else if kind.is_unary()
-            || matches!(
-                kind,
-                TokenKind::Star | TokenKind::DoubleStar | TokenKind::At
-            )
-        {
-            let is_binary = prev_token.map_or(false, |prev_token| {
-                let prev_kind = prev_token.kind();
-
-                // Check if the operator is used as a binary operator.
-                // Allow unary operators: -123, -x, +1.
-                // Allow argument unpacking: foo(*args, **kwargs)
-                // Allow decorators: @foo, @foo(1)
-                matches!(
-                    prev_kind,
-                    TokenKind::Rpar | TokenKind::Rsqb | TokenKind::Rbrace
-                ) || !(prev_kind.is_operator()
-                    || prev_kind.is_keyword()
-                    || prev_kind.is_soft_keyword())
-            });
-
-            if is_binary {
-                if kind == TokenKind::DoubleStar {
-                    // Enforce consistent spacing, but don't enforce whitespaces.
-                    NeedsSpace::Optional
+        if let Some(prev_token) = prev_token {
```

Comment:
I put this whole thing in an `if-let` which causes an enormous diff. It felt cleaner than if `if-else`.

Reply:
URL for review https://github.com/charliermarsh/ruff/pull/4458/files?w=1